File: myTrial-api/src/main/java/com/cmpe138/mytrial/service/PatientService.java
// SJSU CMPE 138 Fall 2021 TEAM1
package com.cmpe138.mytrial.service;

import java.util.List;

import com.cmpe138.mytrial.model.Patient;

public interface PatientService {

	List<Patient> getAll();

	List<Patient> getPatientByResearcherId(String researcher_id);

	/**
	 * For manager using, can get any patient using patient id
	 * 
	 * @param patient_id
	 * @return
	 */
	Patient getPatientById(String patient_id);

	/**
	 * For researcher, can get patients under his patient list
	 * 
	 * @param researcher_id
	 * @param patient_id
	 * @return
	 */
	Patient getPatientById(String researcher_id, String patient_id);

	/**
	 * @param username
	 * @param password
	 * @return
	 */
	Patient getPatientByUsernamePassword(String username, String password);

	void updatePatientUsingPatientID(String disease, String phase, String status, String trial_id, String patient_id);

	List<Patient> getPatientByTrialId(String trial_id);

	/**
	 * For researcher, can delete a patient by patient ID
	 * 
	 * @param patient_id
	 * @return
	 */
	boolean deletePatientById(int patient_id);
}

File: mytrial-fe/src/service/PatientService.js
// SJSU CMPE 138 Fall 2021 TEAM1
import axios from 'axios';
import { PATIIENT_API_BASE_URL } from '../settings/index'

class PatientService {
    // config has researcher_id
    getPatients(config) {
        return axios.get(PATIIENT_API_BASE_URL, config);
    }

    createPatient(patient) {
        return axios.post(PATIIENT_API_BASE_URL, patient);
    }

    getPatientById(patientId, config) {
        return axios.get(PATIIENT_API_BASE_URL + '/' + patientId, config);
    }

    updatePatient(patient, patientId) {
        return axios.put(PATIIENT_API_BASE_URL + '/' + patientId, patient);
    }

    deletePatient(patientId) {
        return axios.delete(PATIIENT_API_BASE_URL + '/' + patientId);
    }
}

export default new PatientService()
